vmm/src/config.rs

```diff
@@ -1252,6 +1257,23 @@ impl VmConfig {
                 if disk.vhost_socket.as_ref().and(disk.path.as_ref()).is_some() {
                     return Err(ValidationError::DiskSocketAndPath);
                 }
+                if disk.vhost_user && !self.memory.shared {
+                    return Err(ValidationError::VhostUserRequiresSharedMemory);
+                }
+            }
+        }
+
+        if let Some(nets) = &self.net {
+            for net in nets {
+                if net.vhost_user && !self.memory.shared {
+                    return Err(ValidationError::VhostUserRequiresSharedMemory);
+                }
+            }
+        }
+
+        if let Some(fses) = &self.fs {
+            if !fses.is_empty() {
```

Comment:
Oh okay so you're just returning an error if there's a virtio-fs device :rofl: 
I think you meant `if !fses.is_empty() && !self.memory.shared`.

Reply:
Ah, yes, clippy was nagging me about my earlier code and I "optimised" it too much :-)

Reply:
Hehe :)

Reply:
Fixed now!